sklearn/metrics/_classification.py

```diff
@@ -732,7 +732,7 @@ def jaccard_score(
         labels are column indices. By default, all labels in ``y_true`` and
         ``y_pred`` are used in sorted order.
 
-    pos_label : str or int, default=1
+    pos_label : int, float, bool or str, default=1
```

Comment:
It's unfortunate that `pos_label` can be `float`. I think it's too confusing to make `float` a hidden option, so documenting it here is okay with me.

Do you think we can get to a point where we can reject float arrays as classification targets? It likely breaks too much code to be worth it.

Reply:
Yep, this is unfortunate.

I know that for the parameter validation, we saw that we introduced a regression via a failing and the failing test was not just passing some array but more a use case. But I don't recall what was it.